<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>xrf_detector_widget</class>
 <widget class="QWidget" name="xrf_detector_widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1071</width>
    <height>759</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="baseSize">
   <size>
    <width>100</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="detector_name_layout">
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QLabel" name="detector_name_label">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <pointsize>12</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Tardis</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="dwell_time_layout">
       <property name="spacing">
        <number>3</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>Dwell Time:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="PyDMLineEdit" name="PyDMLineEdit">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="toolTip">
          <string>The wall time (in seconds) for acquiring each frame. Applies to both setup and scanning.</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="placeholderText">
          <string>Dwell time (s)</string>
         </property>
         <property name="showUnits" stdset="0">
          <bool>true</bool>
         </property>
         <property name="channel" stdset="0">
          <string>haven://${DEV}.dwell_time</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="PyDMCheckbox" name="continuous_checkbox">
       <property name="toolTip">
        <string>Acquire frames continuously and display them on the plot.</string>
       </property>
       <property name="text">
        <string>Continuous</string>
       </property>
       <property name="channel" stdset="0">
        <string>haven://${DEV}.acquire_multiple</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="PyDMPushButton" name="oneshot_button">
       <property name="toolTip">
        <string>Collect a single frame to show on the plot.</string>
       </property>
       <property name="text">
        <string>Acquire</string>
       </property>
       <property name="channel" stdset="0">
        <string>haven://${DEV}.acquire_single</string>
       </property>
       <property name="pressValue" stdset="0">
        <string>1</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="accumulate_checkbox">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>If checked, successive frames will be added together, if unchecked, each acquisition will be a new frame.</string>
       </property>
       <property name="text">
        <string>Accumulate</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>6</number>
       </property>
       <item>
        <widget class="PyDMLabel" name="PyDMLabel_35">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="toolTip">
          <string>Currest status of the detector.</string>
         </property>
         <property name="text">
          <string>Acquiring</string>
         </property>
         <property name="precision" stdset="0">
          <number>0</number>
         </property>
         <property name="showUnits" stdset="0">
          <bool>false</bool>
         </property>
         <property name="precisionFromPV" stdset="0">
          <bool>true</bool>
         </property>
         <property name="alarmSensitiveContent" stdset="0">
          <bool>false</bool>
         </property>
         <property name="alarmSensitiveBorder" stdset="0">
          <bool>true</bool>
         </property>
         <property name="PyDMToolTip" stdset="0">
          <string/>
         </property>
         <property name="channel" stdset="0">
          <string>haven://${DEV}.acquire</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_17">
         <property name="text">
          <string>(</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="PyDMLabel" name="PyDMLabel_33">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="toolTip">
          <string>Number of frames captured in the current acquisition.</string>
         </property>
         <property name="styleSheet">
          <string notr="true">padding: 0;</string>
         </property>
         <property name="text">
          <string>0</string>
         </property>
         <property name="textInteractionFlags">
          <set>Qt::TextSelectableByMouse</set>
         </property>
         <property name="channel" stdset="0">
          <string>haven://${DEV}.cam.array_counter</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>)</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="Line" name="line_3">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="total_layout">
       <property name="spacing">
        <number>3</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="label_4">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Total:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="PyDMLabel" name="total_counts_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Monospace</family>
           <pointsize>9</pointsize>
           <weight>50</weight>
           <bold>false</bold>
          </font>
         </property>
         <property name="toolTip">
          <string>Total number of detection events over all elements.</string>
         </property>
         <property name="styleSheet">
          <string notr="true">padding: 0;</string>
         </property>
         <property name="text">
          <string>N/A</string>
         </property>
         <property name="channel" stdset="0">
          <string>haven://${DEV}.total_count</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="dead_time_layout">
       <property name="spacing">
        <number>3</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="label_6">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Dead Time:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="PyDMLabel" name="deadtime_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="toolTip">
          <string>Overall dead time (in per-cent) across all elements for the last acquisition.</string>
         </property>
         <property name="styleSheet">
          <string notr="true">padding: 0;</string>
         </property>
         <property name="text">
          <string>22.8</string>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <property name="precision" stdset="0">
          <number>2</number>
         </property>
         <property name="precisionFromPV" stdset="0">
          <bool>false</bool>
         </property>
         <property name="channel" stdset="0">
          <string>haven://${DEV}.dead_time_average</string>
         </property>
         <property name="displayFormat" stdset="0">
          <enum>PyDMLabel::Decimal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="text">
          <string>%</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="dead_time_range_layout">
       <property name="spacing">
        <number>2</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="label_11">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>(</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="PyDMLabel" name="dead_time_min_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="toolTip">
          <string>Minimum dead time (in per-cent) across all elements for the last acquisition.</string>
         </property>
         <property name="styleSheet">
          <string notr="true">padding: 0;</string>
         </property>
         <property name="text">
          <string>14.0</string>
         </property>
         <property name="PyDMToolTip" stdset="0">
          <string>Lowest dead time across all elements.</string>
         </property>
         <property name="channel" stdset="0">
          <string>haven://${DEV}.dead_time_min</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_15">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="text">
          <string>%</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_12">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>−</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="PyDMLabel" name="dead_time_max_label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="toolTip">
          <string>Maximum dead time (in per-cent) across all elements for the last acquisition.</string>
         </property>
         <property name="styleSheet">
          <string notr="true">padding: 0;</string>
         </property>
         <property name="text">
          <string>24.0</string>
         </property>
         <property name="PyDMToolTip" stdset="0">
          <string>Highest dead time across all elements.</string>
         </property>
         <property name="channel" stdset="0">
          <string>haven://${DEV}.dead_time_max</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_16">
         <property name="font">
          <font>
           <family>Monospace</family>
          </font>
         </property>
         <property name="text">
          <string>%</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_13">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>)</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="elements_tab">
       <attribute name="title">
        <string>Elements</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QSplitter" name="splitter_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <widget class="QWidget" name="verticalLayoutWidget_4">
           <layout class="QVBoxLayout" name="elements_controls_layout">
            <item>
             <widget class="QScrollArea" name="scrollArea_2">
              <property name="horizontalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOff</enum>
              </property>
              <property name="widgetResizable">
               <bool>true</bool>
              </property>
              <widget class="QWidget" name="scrollAreaWidgetContents_2">
               <property name="geometry">
                <rect>
                 <x>0</x>
                 <y>0</y>
                 <width>896</width>
                 <height>428</height>
                </rect>
               </property>
               <layout class="QGridLayout" name="gridLayout">
                <item row="0" column="0">
                 <layout class="QGridLayout" name="mcas_layout">
                  <property name="topMargin">
                   <number>0</number>
                  </property>
                  <item row="0" column="0">
                   <widget class="QLabel" name="label">
                    <property name="font">
                     <font>
                      <weight>75</weight>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>Element</string>
                    </property>
                   </widget>
                  </item>
                  <item row="0" column="1">
                   <widget class="QLabel" name="label_2">
                    <property name="font">
                     <font>
                      <weight>75</weight>
                      <bold>true</bold>
                     </font>
                    </property>
                    <property name="text">
                     <string>Counts</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="0">
                   <widget class="QLabel" name="label_5">
                    <property name="text">
                     <string>Total</string>
                    </property>
                   </widget>
                  </item>
                  <item row="1" column="1">
                   <widget class="QLabel" name="total_counts_label_2">
                    <property name="text">
                     <string>##########</string>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item row="3" column="0">
                 <spacer name="verticalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="MCAPlotWidget" name="mca_plot_widget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PyDMLabel</class>
   <extends>QLabel</extends>
   <header>pydm.widgets.label</header>
  </customwidget>
  <customwidget>
   <class>PyDMCheckbox</class>
   <extends>QCheckBox</extends>
   <header>pydm.widgets.checkbox</header>
  </customwidget>
  <customwidget>
   <class>PyDMLineEdit</class>
   <extends>QLineEdit</extends>
   <header>pydm.widgets.line_edit</header>
  </customwidget>
  <customwidget>
   <class>PyDMPushButton</class>
   <extends>QPushButton</extends>
   <header>pydm.widgets.pushbutton</header>
  </customwidget>
  <customwidget>
   <class>MCAPlotWidget</class>
   <extends>QWidget</extends>
   <header>firefly.xrf_detector</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>continuous_checkbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>oneshot_button</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>223</x>
     <y>152</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
